packages/core/src/agent/MessageReceiver.ts

```diff
@@ -131,53 +127,65 @@ export class MessageReceiver {
       this.transportService.saveSession(session)
     }
 
+    const messageContext = new InboundMessageContext(message, {
+      // Only make the connection available in message context if the connection is ready
+      // To prevent unwanted usage of unready connections. Connections can still be retrieved from
+      // Storage if the specific protocol allows an unready connection to be used.
+      connection: connection?.isReady ? connection : undefined,
+      senderVerkey: senderKey,
+      recipientVerkey: recipientKey,
```

Comment:
```suggestion
      senderKey,
      recipientKey,
```

I see you updated verkey to key in a lot of places, maybe would be good to also remove the reference to verkey here.

However we do lose some context here, as a verkey is a base58 encoded public key (probably ed25519) while key is more general and doesn't specify the encoding of the key.

For now lets use `senderKey` etc but it may be good to use something like `senderKeyBase58`

Reply:
That's interesting... Do you mean that the sender key differs from the recipient keys? I see that `Wallet`, for example, contains `recipientKeys: string[], senderVerkey?: string`. Is the naming on purpose to distinguish different formats?

Reply:
Ah, I just realized that it should be the same because we're using the sender key as recipient and vice-versa. I think if it's currently the same format across the framework we can name it as `xxxKey`. But you have a good point, it can be eventually a different key format and we should distinguish between them.